### PR TITLE
Updating to use async redux functions

### PR DESCRIPTION
**Pull request by Azero123**
#### Summary
When updating to use Flow many functions became async and these changes are needed to prevent issues for the webapp.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/9771

#### Related Pull Requests
- Has redux changes
https://github.com/mattermost/mattermost-redux/pull/849

### CODE REVIEW

File: components/edit_post_modal/edit_post_modal.test.jsx [hunk 1]
```diff
@@ -36,10 +36,10 @@ function createEditPost({canEditPost, canDeletePost, ctrlSend, config, license,
         title: 'test',
     };
     const actionsProp = actions || {
-        editPost: jest.fn(),
-        addMessageIntoHistory: jest.fn(),
-        hideEditPostModal: jest.fn(),
-        openModal: jest.fn(),
+        editPost: jest.fn((data) => new Promise((resolve) => resolve(data))),
```
Comment (hmhealey) [inline components/edit_post_modal/edit_post_modal.test.jsx]: These can be simplified by using `Promise.resolve`
```suggestion
        editPost: jest.fn((data) => Promise.resolve(data)),
```
Or possibly even
```suggestion
        editPost: jest.fn(Promise.resolve),
```

**Comment by Azero123** [comment]
@hmhealey Thanks for the review I applied the changes, please let me know if there is more I can do (:

**Comment by hanzei** [comment]
@sudheerDev Gentle reminder about reviewing this PR

**Comment by sudheerDev** [comment]
LGTM but waiting on the other PR as it might cause some more changes to this PR

**Comment by hanzei** [comment]
Closing together with https://github.com/mattermost/mattermost-redux/pull/849 due to interactivity.

**Pull request closed without merging**